manta-crypto/src/eclair/num.rs

```diff
@@ -180,6 +257,90 @@ where
     }
 }
 
+impl<T, const BITS: usize, COM> Zero<COM> for UnsignedInteger<T, BITS>
+where
+    T: Zero<COM>,
+{
+    type Verification = T::Verification;
+
+    #[inline]
+    fn zero(compiler: &mut COM) -> Self {
+        Self::new_unchecked(T::zero(compiler))
+    }
+
+    #[inline]
+    fn is_zero(&self, compiler: &mut COM) -> Self::Verification {
+        self.0.is_zero(compiler)
+    }
+}
+
+impl<T, const BITS: usize, COM> One<COM> for UnsignedInteger<T, BITS>
+where
+    T: One<COM>,
+{
+    type Verification = T::Verification;
+
+    #[inline]
+    fn one(compiler: &mut COM) -> Self {
+        Self::new_unchecked(T::one(compiler))
+    }
+
+    #[inline]
+    fn is_one(&self, compiler: &mut COM) -> Self::Verification {
+        self.0.is_one(compiler)
+    }
+}
+
+impl<T, const BITS: usize, COM> PartialEq<Self, COM> for UnsignedInteger<T, BITS>
+where
+    COM: Has<bool>,
+    T: PartialEq<T, COM>,
+{
+    #[inline]
+    fn eq(&self, rhs: &Self, compiler: &mut COM) -> Bool<COM> {
+        self.0.eq(&rhs.0, compiler)
+    }
+
+    #[inline]
+    fn ne(&self, rhs: &Self, compiler: &mut COM) -> Bool<COM>
+    where
+        Bool<COM>: Not<COM, Output = Bool<COM>>,
+    {
+        self.0.ne(&rhs.0, compiler)
+    }
+
+    #[inline]
+    fn assert_equal(&self, rhs: &Self, compiler: &mut COM)
+    where
+        COM: Assert,
+    {
+        self.0.assert_equal(&rhs.0, compiler)
+    }
+}
+
+impl<T, const BITS: usize, COM> ConditionalSelect<COM> for UnsignedInteger<T, BITS>
+where
+    COM: Has<bool>,
+    T: ConditionalSelect<COM>,
+{
+    #[inline]
+    fn select(bit: &Bool<COM>, true_value: &Self, false_value: &Self, compiler: &mut COM) -> Self {
+        Self::new_unchecked(T::select(bit, &true_value.0, &false_value.0, compiler))
```

Comment:
This is a separate case as `zero` and `one` which are debatable whether they should be checked or not. This can be completely unchecked since the values `true_value` and `false_value` are already checked.